react/SellerWrapper.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useCallback, useEffect, useRef } from 'react'
+import { useProduct, useProductDispatch } from 'vtex.product-context'
+import { useCssHandles } from 'vtex.css-handles'
+import type { Item } from 'vtex.product-context/react/ProductTypes'
+
+import useSeller from './hooks/useSeller'
+
+const CSS_HANDLES = ['sellerWrapper']
+const SELECT_ITEM_EVENT = 'SET_SELECTED_ITEM'
+
+type SellerWrapperProps = {
+  children: React.ReactNode
+}
+
+const SellerWrapper = ({ children }: SellerWrapperProps) => {
+  const { seller } = useSeller()
+  const dispatch = useProductDispatch()
+  const { selectedItem, product } = useProduct() ?? {}
+  const latestItem = useRef((null as unknown) as Item)
+  const handles = useCssHandles(CSS_HANDLES)
+
+  const memoizedCallback = useCallback(
```

Comment:
It would be clearer if this function was named after what it does, instead of just indicating that it's a callback. For example: `addSellerDefaultToItem`.